src/dbus/mutter_display_config.rs

```diff
@@ -316,11 +332,102 @@ impl DisplayConfig {
 
     #[zbus(signal)]
     pub async fn monitors_changed(ctxt: &SignalEmitter<'_>) -> zbus::Result<()>;
+
+    async fn apply_monitors_config(
+        &self,
+        _serial: u32,
+        method: u32,
+        logical_monitor_configs: Vec<LogicalMonitorConfiguration>,
+        _properties: HashMap<String, OwnedValue>,
+    ) -> fdo::Result<()> {
+        let current_conf = self.ipc_outputs.lock().unwrap();
+        let mut messages = HashMap::new();
+        for requested_config in logical_monitor_configs.into_iter() {
+            for (connector, mode, _props) in requested_config.monitors {
+                if current_conf
+                    .values()
+                    .find(|o| o.name == connector)
+                    .is_none()
+                {
+                    return Err(zbus::fdo::Error::Failed(format!(
+                        "Connector '{}' not found",
+                        connector
+                    )));
+                }
+                let msgs = vec![
```

Comment:
In comparison, the wlr-output-management impl creates and sets a complete new output configuration from scratch. Wouldn't that also be more appropriate here?

As far as I understand, the functional difference is that it is applied all at once, without repositioning outputs every step. That said, I think your approach should result in the same outcome anyway? So I guess this code can stay as is.

Reply:
Oh, that's possible? Right, I'm not restricted to ipc related API now that I'm using the channels..

Reply:
Had to extract the find-or-create-config-by-name part from `apply_transient_output_config` but I think it does look better this way, creating the `niri_config::Output` from scratch.

Actually the outcome wasn't *exactly* the same: `apply_transient_output_config` was calling `reload_output_config` after every action…